gdx/src/com/badlogic/gdx/math/Octree.java

```diff
@@ -0,0 +1,307 @@
+
+        this.root = new OctreeNode(realMin, realMax, maxDepth);
+        this.collider = collider;
+    }
+
+    public void add (T aabb) {
+        root.add(aabb, maxItemsPerNode, collider);
+    }
+
+    public void remove (T aabb) {
+        root.remove(aabb);
+    }
+
+    /**
+     * Method to retrieve all the geometries
+     *
+     * @param resultSet
+     * @return the result set
+     */
+    public ObjectSet<T> getAll (ObjectSet<T> resultSet) {
+        root.getAll(resultSet);
+        return resultSet;
+    }
+
+    /**
+     * Method to query geometries inside nodes that the aabb intersects
+     * Can be used as broad phase
+     * @param aabb - The bounding box to query
+     * @param result - Set to be populated with objects inside the BoundingBoxes
+     */
+    public ObjectSet<T> query (BoundingBox aabb, ObjectSet<T> result) {
+        root.query(aabb, result);
+        return result;
+    }
+
+    /**
+     * Method to query geometries inside nodes that the frustum intersects
+     * Can be used as broad phase
+     *
+     * @param frustum - The frustum to query
+     * @param result set populated with objects near from the frustum
+     */
+    public ObjectSet<T> query (Frustum frustum, ObjectSet<T> result) {
+        root.query(frustum, result);
+        return result;
+    }
+
+    public T rayCast (Ray ray) {
+        return rayCast(ray, Float.POSITIVE_INFINITY);
+    }
+
+    public T rayCast (Ray ray, float maxDistance) {
+        RayCastResult<T> result = new RayCastResult<>();
+        result.distance = Float.NEGATIVE_INFINITY;
+        root.rayCast(ray, maxDistance, collider, result);
+        return result.geometry;
+    }
+
+    public void setMaxDepth (int maxDepth) {
+        this.maxDepth = maxDepth;
+        this.root.setLevel(maxDepth);
+    }
+
+    public void setMaxItemsPerNode (int maxItemsPerNode) {
+        this.maxItemsPerNode = maxItemsPerNode;
+    }
+
+    /**
+     * Method to get nodes boxes. Useful for render purpose
+     * @param boxes
+     */
+    public ObjectSet<BoundingBox> getNodesBoxes (ObjectSet<BoundingBox> boxes) {
+        root.getBoundingBox(boxes);
+        return boxes;
+    }
+
+    protected class OctreeNode {
+
+        protected int level;
+
+        // Represents the boundary of the node
+        private BoundingBox bounds;
+
+        private Array<OctreeNode> children;
+        private ObjectSet<T> geometries = null;
+
+        public OctreeNode (Vector3 min, Vector3 max, int level) {
+            this(min.x, min.y, min.z, max.x, max.y, max.z, level);
+        }
+
+        public OctreeNode (float x1, float y1, float z1, float x2, float y2, float z2, int level) {
+            bounds = new BoundingBox(new Vector3(x1, y1, z1), new Vector3(x2, y2, z2));
+            this.level = level;
+        }
+
+        protected void split (int maxItemsPerNode, Collider<T> collider) {
```

Comment:
maxItemsPerNode is not needed here, this information is already accessible from parent class.